src/Router.ts

```diff
@@ -318,11 +319,10 @@ export class Router<C extends Context> extends Evented {
 						//
 						// Reset selected routes if not dispatched from start().
 						this._currentSelection = dispatchFromStart ? result : [];
-
-						for (const { handler, params } of result) {
-							catchRejection(this, context, path, handler({ context, params }));
-						}
-
+						const promise = result.reduce((promise, { handler, params }) => {
```

Comment:
Ah, I see, if one `handler` rejects then all the subsequent ones shouldn't be called?